pkgs/by-name/ba/bartender/package.nix

```diff
@@ -32,17 +39,37 @@ stdenvNoCC.mkDerivation (finalAttrs: {
     runHook postInstall
   '';
 
+  passthru.updateScript = lib.getExe (writeShellApplication {
+    name = "bartender-update-script";
+    runtimeInputs = [
+      curl
+      xmlstarlet
+      common-updater-scripts
+    ];
+    text = ''
+      version_major="${lib.versions.major finalAttrs.version}"
+      url="https://www.macbartender.com/B2/updates/AppcastB$version_major.xml"
+      version=$(curl -s "$url" | xmlstarlet sel -t -v '(//item)[last()]/sparkle:shortVersionString' -n)
+      update-source-version bartender "$version"
+    '';
+  });
+
   meta = {
     description = "Take control of your menu bar";
     longDescription = ''
       Bartender is an award-winning app for macOS that superpowers your menu bar, giving you total control over your menu bar items, what's displayed, and when, with menu bar items only showing when you need them.
       Bartender improves your workflow with quick reveal, search, custom hotkeys and triggers, and lots more.
     '';
     homepage = "https://www.macbartender.com";
-    changelog = "https://www.macbartender.com/Bartender${lib.versions.major finalAttrs.version}/release_notes/";
+    changelog = "https://macbartender.com/B2/updates/${
+      builtins.replaceStrings [ "." ] [ "-" ] finalAttrs.version
+    }/rnotes.html";
     license = [ lib.licenses.unfree ];
     sourceProvenance = [ lib.sourceTypes.binaryNativeCode ];
-    maintainers = with lib.maintainers; [ stepbrobd DimitarNestorov ];
-    platforms = [ "aarch64-darwin" "x86_64-darwin" ];
+    maintainers = with lib.maintainers; [
+      stepbrobd
+      DimitarNestorov
+    ];
+    platforms = lib.platforms.darwin;
```

Comment:
```nix
nix-repl> lib.platforms.darwin
[
  "x86_64-darwin"
  "i686-darwin"
  "aarch64-darwin"
  "armv7a-darwin"
]
```

I don't think bartender supports i686-darwin and armv7a-darwin

Reply:
Yea, but on nix search page it just shows x86 and arch64 only, plus there's no Apple PC devices that have these architectures...

This is more for syntax sugar than anything else

Reply:
idk what the current convention is, if `lib.platforms.darwin` is preferred over explicit listings in other darwin packages then feel free to ignore this

Reply:
Idk if there's any convention, but from a POV, it's "cleaner" to have `lib.platforms.darwin`. As well, it's impossible to run Nix on `armv7a-darwin` (32-bit iPhones?) or `i686-darwin` (OS X 10.3?), so it's impossible to have a situation where someone will be trying to build Bartender on those platforms.

Reply:
I'd say this is fine, especially as I don't think anything *technically* stops it? I've also only rarely seen an explicit listing of x86_64/aarch64-darwin

Maybe use [meta.badPlatforms](https://nixos.org/manual/nixpkgs/stable/#var-meta-badPlatforms)?